### PR TITLE
remove dead code

### PR DESCRIPTION
**Pull request by shiywang**
Fixing https://gitlab.cee.redhat.com/openshift-art/tools/elliott/merge_requests/78/diffs#note_875105

### CODE REVIEW

**Comment by vfreex** [comment]
Good job. I think those for validating ocp-build-data should also be removed since they are outdated and  handled by ocp-build-data-validator

**Comment by vfreex** [comment]
@shiywang Can you rebase this PR?

File: elliottlib/container.py [hunk 1]
```diff
@@ -1,62 +0,0 @@
-import shlex
-import exectools
-import logutil
-import constants
-
-logger = logutil.getLogger(__name__)
-
-
-class DockerContainer(object):
```
Comment (sosiouxme) [inline elliottlib/container.py]: This _might_ still prove useful for what we were doing with it before (basically, starting containers to run orphan checks). But that's no reason to keep this around, we can resurrect it from git if needed.

File: elliottlib/runtime.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 import assertion
 import exectools
 from pushd import Dir
-from image import ImageMetadata
+from imagecfg import ImageMetadata
```
Comment (sosiouxme) [inline elliottlib/runtime.py]: should the `ImageMetadata` be removed from `elliottlib/image.py` then?

File: elliottlib/imagecfg.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    @property
+    def base_only(self):
+        """
+        Some images are marked base-only.  Return the flag from the config file
+        if present.
+        """
+        return self.config.base_only
+
+    def get_latest_build_info(self):
```
Comment (sosiouxme) [inline elliottlib/imagecfg.py]: This has recently been updated and now takes the `product_version` param. Start with updated `image.py` -- this breaks image finds:
```
🐚 elliott -g openshift-4.1 find-builds -k image
2019-10-31 04:34:18,179 INFO Cloning config data from https://github.com/openshift/ocp-build-data.git
2019-10-31 04:34:18,678 INFO Using branch from group.yml: rhaos-4.1-rhel-7
Generating list of images: Hold on a moment, fetching Brew buildinfo
[********************************************************************************************************************]
[******Traceback (most recent call last):
...
  File "/home/lmeyer/.local/lib/python2.7/site-packages/elliottlib/cli/find_builds_cli.py", line 112, in find_builds_cli
    unshipped_builds = _fetch_builds_by_kind_image(runtime, product_version, session)
  File "/home/lmeyer/.local/lib/python2.7/site-packages/elliottlib/cli/find_builds_cli.py", line 187, in _fetch_builds_by_kind_image
    lambda build: build.get_latest_build_info(product_version_overide)
  File "/home/lmeyer/.local/lib/python2.7/site-packages/elliottlib/util.py", line 229, in parallel_results_with_progress
    inputs)
  File "/usr/lib64/python2.7/multiprocessing/pool.py", line 253, in map
    return self.map_async(func, iterable, chunksize).get()
  File "/usr/lib64/python2.7/multiprocessing/pool.py", line 572, in get
    raise self._value
TypeError: get_latest_build_info() takes exactly 1 argument (2 given)
```

File: elliottlib/runtime.py (hunk 2 above)
Comment (shiywang) [reply elliottlib/runtime.py]: My intention was rename `image.py` -> `imagecfg.py` to consist with `rpmcfg.py` 
and `ImageMetadata` is still in use here: `meta = ImageMetadata(self, data_obj)`

File: elliottlib/container.py (hunk 1 above)
Comment (shiywang) [reply elliottlib/container.py]: Sounds good.

File: elliottlib/imagecfg.py (hunk 3 above)
Comment (shiywang) [reply elliottlib/imagecfg.py]: Fixed

**Comment by thiagoalessio** [comment]
Holy Moly @shiywang that's a LOT of dead code. thanks for cleaning up the house!
I see that you addressed Luke's comments, our (few) tests are still passing.

I'll /lgtm and merge it, because would be nice to have this week's release manager already using your changes to see if we encounter any edge case.
